control-plane/agents/src/bin/core/controller/reconciler/node/snapshot.rs

```diff
@@ -0,0 +1,99 @@
+use crate::controller::{
+    io_engine::ReplicaSnapshotApi,
+    reconciler::poller::ReconcilerWorker,
+    resources::{OperationGuardArc, ResourceUid},
+    task_poller::{PollContext, PollResult, PollTimer, PollerState, TaskPoller},
+};
+use stor_port::types::v0::{
+    store::node::NodeSpec,
+    transport::{DestroyReplicaSnapshot, ListReplicaSnapshots, SnapshotId},
+};
+
+/// Node snapshot garbage collector.
+#[derive(Debug)]
+pub(super) struct NodeSnapshotGarbageCollector {
+    counter: PollTimer,
+}
+impl NodeSnapshotGarbageCollector {
+    /// Return a new `Self`.
+    pub(super) fn new() -> Self {
+        Self {
+            counter: ReconcilerWorker::garbage_collection_period(),
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl TaskPoller for NodeSnapshotGarbageCollector {
+    async fn poll(&mut self, context: &PollContext) -> PollResult {
+        let nodes = context.specs().nodes();
+        let mut results = Vec::with_capacity(nodes.len());
+
+        for node in nodes {
+            let mut node_spec = context.registry().specs().guarded_node(node.id()).await?;
+            results.push(delete_unknown_volume_snapshot_reconciler(context, &mut node_spec).await);
+        }
+        Self::squash_results(results)
+    }
+
+    async fn poll_timer(&mut self, _context: &PollContext) -> bool {
+        self.counter.poll()
+    }
+}
+
+/// Identify unknown snapshots and attempt to clean them up.
+#[tracing::instrument(skip(node_spec, context), level = "trace", fields(request.reconcile = true))]
+async fn delete_unknown_volume_snapshot_reconciler(
+    context: &PollContext,
+    node_spec: &mut OperationGuardArc<NodeSpec>,
+) -> PollResult {
+    let node_wrapper = match context.registry().node_wrapper(node_spec.uid()).await {
+        Ok(node_wrapper) => node_wrapper,
+        Err(error) => return PollResult::Err(error),
+    };
```

Comment:
Can't we do just this?

```suggestion
    let node_wrapper = context.registry().node_wrapper(node_spec.uid()).await?;
```